### ISSUE TITLE
tf.debugging.assert_type throws error when checking type of RaggedTensor

### ISSUE THREAD

**Issue by abhmul** [tensorflow/tensorflow#53466]
<em>Please make sure that this is a bug. As per our
[GitHub Policy](https://github.com/tensorflow/tensorflow/blob/master/ISSUES.md),
we only address code/doc bugs, performance issues, feature requests and
build/installation issues on GitHub. tag:bug_template</em>

**System information**
- Have I written custom code (as opposed to using a stock example script provided in TensorFlow): **Yes**
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04): **Ubuntu 18.04.6 LTS**
- Mobile device (e.g. iPhone 8, Pixel 2, Samsung Galaxy) if the issue happens on a mobile device: **N/A**
- TensorFlow installed from (source or binary): **binary**
- TensorFlow version (use command below): **v2.7.0-rc1-69-gc256c071bb2 2.7.0**
- Python version: **3.9.0**
- Bazel version (if compiling from source): **N/A**
- GCC/Compiler version (if compiling from source): **N/A**
- CUDA/cuDNN version: **N/A**
- GPU model and memory: **N/A**

You can collect some of this information using our environment capture
[script](https://github.com/tensorflow/tensorflow/tree/master/tools/tf_env_collect.sh)
You can also obtain the TensorFlow version with:
1. TF 1.0: `python -c "import tensorflow as tf; print(tf.GIT_VERSION, tf.VERSION)"`
2. TF 2.0: `python -c "import tensorflow as tf; print(tf.version.GIT_VERSION, tf.version.VERSION)"`

**Describe the current behavior**
When calling `tf.debugging.assert_type` on a `RaggedTensor`, a `ValueError: TypeError: object of type 'RaggedTensor' has no len()` is thrown.

**Describe the expected behavior**
The assertion check should pass if the ragged tensor is the correct type or fail otherwise.

**[Contributing](https://www.tensorflow.org/community/contribute)**

- Do you want to contribute a PR? (yes/no): **no**
- Briefly describe your candidate solution(if contributing): **N/A**

**Standalone code to reproduce the issue**
Provide a reproducible test case that is the bare minimum necessary to generate
the problem. If possible, please share a link to Colab/Jupyter/any notebook.

```python
import tensorflow as tf

a = tf.ragged.constant([[1, 2], [1]])
tf.debugging.assert_type(a, tf_type=tf.int32)  # Should pass
>>> ValueError: TypeError: object of type 'RaggedTensor' has no len()
```

**Other info / logs** Include any logs or source code that would be helpful to
diagnose the problem. If including tracebacks, please include the full
traceback. Large logs and files should be attached.

**Workaround**
```
import tensorflow as tf

a = tf.ragged.constant([[1, 2], [1]])
assert a.dtype == tf.int32
```

**Comment by abhmul** [comment]
Is there a larger issue capturing methods that have not been updated to support ragged tensors?

**Comment by mohantym** [comment]
Hi @abhmul ! According to this [document](https://www.tensorflow.org/api_docs/python/tf/debugging/assert_type#args), Supported tensor types are -A Tensor, SparseTensor or tf.Variable . Attaching resolved [gist](https://colab.sandbox.google.com/gist/mohantym/08e1b9b15481d5281ccd5004f8ec1338/github_53466.ipynb#scrollTo=l4DULmprhfKW) for reference. Thanks!

**Comment by abhmul** [comment]
Could we make this a feature reqeust? I don't see the option for me to change the labels.

**Comment by mohantym** [comment]
@chunduriv ! Could you please look at this feature request?

**Comment by jvishnuvardhan** [comment]
@abhmul I think this was resolved in recent `tf-nightly` (2.9.0-dev20220308). [Here](https://colab.research.google.com/gist/jvishnuvardhan/f666cbed13a93cdf656f39abf9d204fc/untitled1180.ipynb) is a gist for reference. Thanks

I am closing this issue as this was resolved. Please feel free to reopen if the issue persists again. Thanks!
